src/gui/log/logmodel.cpp

```diff
@@ -83,38 +91,24 @@ QVariant LogModel::data(const QModelIndex &index, const int role) const
 
 void LogModel::appendLine(const Log::Msg &msg)
 {
-    const QDateTime time = QDateTime::fromMSecsSinceEpoch(msg.timestamp);
-    const QString text = QString("%1 - %2").arg(time.toString(Qt::SystemLocaleShortDate), msg.message);
     beginInsertRows(QModelIndex(), 0, 0);
-    m_items.append({text, msg.type});
+    m_items.second = msg.id;
+    if (m_items.first < 0)
+        m_items.first = msg.id;
```

Comment:
I would use `std::max`

Reply:
That's not the same. A negative value is an invalid ID and indicates an uninitialized log buffer. After the first message, `first` should be less than `second`.